lib/email/receiver.rb

```diff
@@ -444,6 +446,17 @@ def trim_discourse_markers(reply)
     end
 
     def parse_from_field(mail)
+      if mail.bounced? || verp
+        final_recipient = mail.final_recipient
+        return extract_from_address_and_name(final_recipient) if final_recipient.is_a? String
+
+        if final_recipient.is_a? Array
+          final_recipient.each do |from|
```

Comment:
Why loop? It looks like it's always returning the first recipient.

Reply:
nope. I received an array of recipients in [corresponding rspec test](https://github.com/discourse/discourse/blob/c3bc11b6b9185b6373fde1e02623686ebde4feee/spec/components/email/receiver_spec.rb#L133).